### PR TITLE
Fill tests for ruma_events_macros

### PR DESCRIPTION
**Pull request by ldm0**
Part of #31 solving.

### CODE REVIEW

File: ruma-events-macros/tests/ruma_events_macros.rs [hunk 1]
```diff
@@ -233,173 +233,209 @@ impl Display for InvalidEvent {
 }
 
 // See note about wrapping macro expansion in a module from `src/lib.rs`
-pub mod common_case {
-    use std::convert::TryFrom;
-
+#[cfg(test)]
```
Comment (jplatte) [inline ruma-events-macros/tests/ruma_events_macros.rs]: `#[cfg(test)]` is redundant in files inside `tests`.

File: src/lib.rs [hunk 2]
```diff
@@ -789,73 +789,73 @@ impl From<Algorithm> for String {
     }
 }
 
+#[rustfmt::skip]
```
Comment (jplatte) [inline src/lib.rs]: Please don't do this, when it is used on such a large scale, formatting will degrade over time.

File: src/lib.rs [hunk 3]
```diff
@@ -789,73 +789,73 @@ impl From<Algorithm> for String {
...
 #[cfg(test)]
-mod tests {
+mod event_type_tests {
```
Comment (jplatte) [inline src/lib.rs]: Why did you rename the `tests` module?

File: ruma-events-macros/tests/ruma_events_macros.rs (hunk 1 above)
Comment (ldm0) [reply ruma-events-macros/tests/ruma_events_macros.rs]: Oh sorry I didn't realize that.

File: src/lib.rs (hunk 2 above)
Comment (ldm0) [reply src/lib.rs]: I think the 80 width is to small. and will make the test below less readable. I'll revert that.

File: src/lib.rs (hunk 3 above)
Comment (ldm0) [reply src/lib.rs]: I think we need a more sepecific name of this test module. If name `tests` is used, We will get test result like this:
```
...
test tests::algorithms_serialize_and_deserialize_from_display_form ... ok
test tests::event_types_serialize_and_deserialize_from_display_form ... ok
...
```
Which may not that specific? (I guess?) If that's improper, I will revert that.

File: src/lib.rs (hunk 2 above)
Comment (jplatte) [reply src/lib.rs]: The default max line width is 100, actually. We might change that if it makes the code more readable, but that shouldn't be part of this PR.

Comment (ldm0) [reply src/lib.rs]: Ok I get it. Thanks for reviewing! I have revert this and run the rustfmt.

File: src/lib.rs (hunk 3 above)
Comment (jplatte) [reply src/lib.rs]: I think `mod tests` is a Rust convention, I can't remember a test module named something else. It could be a submodule of another module though. `lib.rs` is already quite big actually. How about this: You could create a new PR moving `EventType` and `Algorithm` to their own module respectively, and `pub use` them from the crate root. Then you can rebase this PR on top of that and avoid further increasing `lib.rs`' size while having clearer test names.

Comment (ldm0) [reply src/lib.rs]: Thanks for pointing that out! Working on it.

**Comment by jplatte** [comment]
Now that the `serde_eq` macro is part of ruma-events, do you want to rebase and clean this up?

**Comment by ldm0** [comment]
Yep! Working on it.

**Comment by ldm0** [comment]
Since the serde_eq macro is in the ruma-event crate and not exported. I just left the macro used in this test as it is.

**Comment by jplatte** [comment]
I think the test macros / functions could be public only for tests if the containg module is:

```rust
#[cfg(not(test)]
mod util;
#[cfg(test)]
pub mod util;
```

Apart from that, it would be nice if these tests would also use `json!` literals.

**Comment by ldm0** [comment]
Ah. That would make things easier. Thanks a lot. Working on it.

**Comment by jplatte** [comment]
@ldm0 actually, that might not work. You'll have to try. I don't know whether the test configuration is active on the main crate when compiling it for external tests.

**Comment by ldm0** [comment]
Yep, that doesn't work.

**Comment by ldm0** [comment]
I can't come up with a good function name. So I use: `serde_json_eq_try_from_raw`.
